tapir/shifts/templates/shifts/shift_calendar_template.html

```diff
@@ -10,56 +10,79 @@
         </div>
     </div>
     {% for week, shifts_by_day in shifts_by_weeks_and_days.items %}
         <div class="card mb-2">
             {% with week_group=week_to_group|dictionary_get:week %}
-                <h5 class="card-header">
-                    KW {{ week|date:"W" }} {% translate 'starting' %} {{ week }},
-                    {% translate 'Week ' %}{{ week_group.name }}
-                </h5>
+                <div class="card-header">
+                    <div class="container-fluid">
+                        <div class="row">
+                            <div class="col-5">
+                                <h5>
+                                    KW {{ week|date:"W" }} {% translate 'starting' %} {{ week }},
+                                    {% translate 'Week ' %}{{ week_group.name }}
+                                    <div>
+                                        <script>
+                                            function onFilter() {
+                                                let url = "{% url "shifts:calendar" %}";
+                                                url += "?date_from=" + document.getElementById("date-from").value;
+                                                url += "&date_to=" + document.getElementById("date-to").value;
+                                                window.location = url;
+                                            }
+                                        </script>
+                                    </div>
+                                </h5>
+                            </div>
+                            <div class="col-7 d-flex justify-content-end gap-2">
+                                <div class="d-flex align-items-baseline gap-2">
+                                    <label for="date-from" class="form-label">{% translate "From" %}</label>
+                                    <input id="date-from"
+                                           type="date"
+                                           value="{{ date_from }}"
+                                           class="form-control"
+                                           style="max-width: 160px" />
+                                </div>
+                                <div class="d-flex align-items-baseline gap-2">
+                                    <label for="date-to" class="form-label">{% translate "To" %}</label>
+                                    <input id="date-to"
+                                           type="date"
+                                           value="{{ date_to }}"
+                                           class="form-control"
+                                           style="max-width: 160px" />
+                                </div>
+                                <button class="{% tapir_button_link %}"
+                                        style="max-height: 45px"
+                                        onclick="onFilter()">{% translate "Filter" %}</button>
```

Comment:
We should add an icon for consistency with the other buttons. On other filters we use `<span class="material-icons">filter_alt</span>`